### DISCUSSION TITLE
cpp/basics/array_and_pointer

### DISCUSSION THREAD

**Discussion by giscus** [imarvinle/CSGuide#48] [Announcements]
# cpp/basics/array_and_pointer

编程指北: 计算机系统学习指南, 操作系统, 计算机网络, C++, Java, 算法, 数据结构

https://csguide.cn/cpp/basics/array_and_pointer.html

**Comment by Bob-maker832** [comment]
在函数中获取数组的字节数：
1：使用数组引用加数组大小(缺点是需要提前知道数组的大小)，对于不是该大小的数组，将运行失败。
int func(int(&arr)[6])
{
	return sizeof(arr);
}
2：使用模版加数组引用。
template<class T>
int func(T& arr)
{
	return sizeof(arr);
}

**Reply by huangjiawei128** [reply]
情况1：对于不是该大小的数组，将直接编译失败
